packages/babel-plugin-transform-async-to-generator/test/fixtures/assumption-ignoreFunctionLength-true/basic/input.mjs

```diff
@@ -0,0 +1,16 @@
+foo(async (x) => {});
+foo(async ([x]) => {});
+foo(async ({ x }) => {});
+
+foo(async function (x) {});
+foo(async function ([x]) {});
+foo(async function ({ x }) {});
+
+foo(async ([]) => {});
+foo(async ({}) => {});
+
+foo(async function ([]) {});
+foo(async function ({}) {});
+
+export default async ([...x]) => {};
+
```

Comment:
There is one more case that we should test:
```js
export default async function (x) {}
```

The function should still be wrapped, because it is a function declaration and it needs to be hoisted.

<details>
<summary>Why?</summary>

Otherwise this code doesn't work, when `a.js` is the entrypoint:

```js
// a.js

import x from "./b.js";

x.then(console.log);

export default async function () { return 1 }
```
```js
// b.js

import f from "./a.js";
export default f();
```

</details>

Reply:
Done. test added.